diff --git a/lib/ObjectSerializer.php b/lib/ObjectSerializer.php
index 0570fc5..05bdf0f 100644
--- a/lib/ObjectSerializer.php
+++ b/lib/ObjectSerializer.php
@@ -165,7 +165,10 @@ class ObjectSerializer
     /**
      * Take value and turn it into a string suitable for inclusion in
      * the parameter. If it's a string, pass through unchanged
-     * If it's a datetime object, format it in ISO8601
+     *
+     * Specific for KillBill:
+     * If it's a boolean, format it as true/false instead of 0/1
+     * If it's a datetime object, format it in Y-m-d
      *
      * @param string|\DateTime $value the value of the parameter
      *
@@ -173,8 +176,10 @@ class ObjectSerializer
      */
     public static function toString($value)
     {
-        if ($value instanceof \DateTime) { // datetime in ISO8601 format
-            return $value->format(\DateTime::ATOM);
+        if ($value instanceof \DateTime) {
+            return $value->format('Y-m-d');
+        } elseif (is_bool($value)) {
+            return $value ? 'true' : 'false';
         } else {
             return $value;
         }
